py4j-python/src/py4j/tests/client_server_test.py

```diff
@@ -317,6 +324,56 @@ def testErrorInPy4J(self):
             # Check that the connection is not broken (refs #265)
             val = client_server.jvm.java.lang.Math.abs(-4)
             self.assertEqual(4, val)
+
```

Comment:
@batterseapower small error here: the line `client_server.shutdown()` was removed from testErrorInPy4J and put at the end of testErrorInPythonCallback. Both tests need this line :-)

This explains why I was getting an error with testErrorInPy4j.